src/borg/archiver/debug_cmd.py

```diff
@@ -265,6 +267,74 @@ def do_debug_id_hash(self, args, repository, manifest):
         print(id.hex())
         return EXIT_SUCCESS
 
+    @with_repository(compatibility=Manifest.NO_OPERATION_CHECK)
+    def do_debug_parse_obj(self, args, repository, manifest):
+        """parse object file into meta dict and data (decrypting, decompressing)"""
+
+        # get the object from id
+        hex_id = args.id
+        try:
+            id = unhexlify(hex_id)
+            if len(id) != 32:  # 256bit
+                raise ValueError("id must be 256bits or 64 hex digits")
+        except ValueError as err:
+            print("object id %s is invalid [%s]." % (hex_id, str(err)))
+            return EXIT_ERROR
+
+        with open(args.cdata, "rb") as f:
+            cdata = f.read()
+
+        repo_objs = manifest.repo_objs
+
+        try:
+            meta, data = repo_objs.parse(id=id, cdata=cdata, decompress=True)
+        except Exception:
+            traceback.print_exc()
+            return EXIT_ERROR
+
+        with open(args.json_path, "w") as f:
+            json.dump(meta, f)
+
+        with open(args.data_path, "wb") as f:
+            f.write(data)
+
+        return EXIT_SUCCESS
+
+    @with_repository(compatibility=Manifest.NO_OPERATION_CHECK)
+    def do_debug_format_obj(self, args, repository, manifest):
+        """format file and metadata into borg object file"""
+
+        # get the object from id
+        hex_id = args.id
+        try:
+            id = unhexlify(hex_id)
+            if len(id) != 32:  # 256bit
+                raise ValueError("id must be 256bits or 64 hex digits")
+        except ValueError as err:
+            print("object id %s is invalid [%s]." % (hex_id, str(err)))
+            return EXIT_ERROR
+
+        with open(args.data_path, "rb") as f:
+            data = f.read()
+
+        with open(args.json_path, "r") as f:
+            meta = json.load(f)
+
+        repo_objs = manifest.repo_objs
+
+        ctype = repo_objs.compressor.ID
+        clevel = repo_objs.compressor.level
+
+        try:
+            data_encrypted = repo_objs.format(id=id, meta=meta, data=data, ctype=ctype, clevel=clevel)
```

Comment:
That's too much unneeded effort (my fault, guess i did not document `.format` well enough):

```
repo_obj = repo_objs.format(id=id, meta=meta, data=data)
```

Guess that is all you need, compress=True is the default and it will use the compressor that is already set up as `repo_objs.compressor` to do the compression and also put ctype/clevel into the metadata.

The other arguments are for the use case when it is called with compress=False (then it will not compress, e.g. because the caller already has compressed data, but then it will need to know the uncompressed size, the ctype, the clevel, etc.).